### PR TITLE
fix race conditions when updating PlanningScene

### PR DESCRIPTION
**Pull request by rhaschke**
@v4hn @davetcoleman
This is a new attempt to fix https://github.com/ros-planning/moveit_ros/issues/442, drawing on reverted PRs https://github.com/ros-planning/moveit_ros/pull/716 https://github.com/ros-planning/moveit_ros/pull/724, and https://github.com/ros-planning/moveit_ros/pull/728.

* This PR goes on top of #63 and needs to be rebased after this was merged.
* I removed `AsyncSpinner` and `CallBackQueue` in `PlanningSceneMonitor` for now. With https://github.com/ros-planning/moveit_ros/pull/717 (#59) this is not required anymore to update the robot state. However `syncSceneUpdates()` relies on them, which is why I had to remove them for now too. It's anyway in discussion, whether this function is useful. I will file a separate PR for this.
* I renamed `CurrentStateMonitor::waitForCurrentState()` and added a deprecated fallback function calling the new method `waitForCompleteState()`. Deprecation should be OK in Kinetic and Jade, but not Indigo. This should be considered when back-porting.

### CODE REVIEW

File: moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/current_state_monitor.h [hunk 1]
```diff
@@ -136,11 +137,18 @@ class CurrentStateMonitor
...
 
-  /** @brief Wait for at most \e wait_time seconds until the complete current state is known. Return true if the full state is known */
-  bool waitForCurrentState(double wait_time) const;
+  /** @brief Wait for at most \e wait_time seconds for a robot state more recent than t */
+  bool waitForCurrentState(const ros::Time t=ros::Time::now(), double wait_time=1) const;
+
+  /** @brief Wait for at most \e wait_time seconds until the complete robot state is known. Return true if the full state is known */
+  bool waitForCompleteState(double wait_time) const;
+  /// replaced by waitForCompleteState, will be removed in L-turtle
+  ROS_DEPRECATED bool waitForCurrentState(double wait_time) const;
```
Comment (davetcoleman) [inline moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/current_state_monitor.h]: ``MOVEIT_``

**Comment by davetcoleman** [comment]
LGTM after reviewing the code but I do not have a setup to verify if these timing changes work

File: moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/current_state_monitor.h (hunk 1 above)
Comment (rhaschke) [inline moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/current_state_monitor.h]: Done

**Comment by v4hn** [comment]
At first glance this looks pretty good. I didn't review though.

Why do you want to make this part of https://github.com/ros-planning/moveit/pull/63 ?

I think I would prefer to discuss and merge them separately in the upstream.

**Comment by rhaschke** [comment]
> Why do you want to make this part of ros-planning#63 ?

This PR cannot be rebased to current kinetic-devel without conflicts. So please, merge ros-planning#63 first. Then, I will create a PR for this one in upstream. Conceptually, they are indeed independent of each other.

**Pull request closed without merging**